File: BubbleSort.java
public class BubbleSorting{  



    public static void main(String[] args) {  
                int arr[] ={3,60,35,2,45,320,5};  

             int n = arr.length;  
        int temp = 0;  
         for(int i=0; i < n; i++){  
                 for(int j=1; j < (n-i); j++){  
                          if(arr[j-1] > arr[j]){  
                                 //swap elements  
                                 temp = arr[j-1];  
                                 arr[j-1] = arr[j];  
                                 arr[j] = temp;  
                         } 
                         } 

        }  

        for(int i=0; i < arr.length; i++){  
            System.out.print(arr[i] + " ");  
    } 


}
}  

File: Sorting Algos/bubble_sort.c
#include<stdio.h>
#include<stdlib.h>
#define SIZE 100

void bubble_sort(int *arr,int n);

int main()
{
 int arr[SIZE];
 int *str;
 printf("Enter size of array : ");
 int n;
 scanf("%d",&n);
 printf("Enter elements of array : ");
 int i;
 for(i=0;i<n;i++)
 scanf("%d",&arr[i]);
 str = arr;
 bubble_sort(str,n);
 return 0;
}

void bubble_sort(int *arr,int n)
{
 int i,j,temp;
 for(i=0;i<n;i++)
  for(j=0;j<n-i-1;j++)
   if(arr[j]>arr[j+1])
    {
       temp = arr[j];
       arr[j] = arr[j+1];
       arr[j+1] = temp;
   }
 for(i=0;i<n;i++)
  printf("%d  ",arr[i]);
 printf("\n");
}
